src/plugins/intel_npu/src/plugin/npuw/util.cpp

```diff
@@ -376,7 +376,7 @@ void ov::npuw::util::gather(const ov::SoPtr<ov::ITensor>& src,
     NPUW_ASSERT(idx->get_element_type() == ov::element::i64);
     NPUW_ASSERT(src_type == ov::element::f16 || src_type == ov::element::f32 || src_type == ov::element::f8e4m3 ||
                 src_type == ov::element::f8e5m2 || src_type == ov::element::f8e8m0 || src_type == ov::element::i8 ||
-                src_type == ov::element::u8);
+                src_type == ov::element::u8 || src_type == ov::element::i4 || src_type == ov::element::u4);
```

Comment:
do you still gather from 4 bit? didn't we unpack this 4 bit vocab to fp16 before?